vispy/visuals/tests/test_volume.py

```diff
@@ -249,12 +249,15 @@ def test_set_data_does_not_change_input():
         V.set_data(vol_copy, clim=(0, 200))
         assert np.allclose(vol, vol_copy)
 
-    # for those using float32 who want to avoid the copy operation,
-    # using set_data() with `copy=False` should be expected to alter the data.
-    vol2 = np.array(vol, dtype='float32', copy=True)
+    # dtype has to be the same as the one used to init the texture, or it will
+    # be first coerced to the same dtype as the init
+
+    vol2 = np.array(vol, dtype=np.float32, copy=True)
     assert np.allclose(vol, vol2)
+    # we explicitly create a copy when data would be altered by the texture,
+    # no matter what the user asks, so the data outside should never change
     V.set_data(vol2, clim=(0, 200), copy=False)
-    assert not np.allclose(vol, vol2)
+    assert np.allclose(vol, vol2)
```

Comment:
This test actually used to *rely* on the "broken" behaviour that we just changed, so I simply inverted the test.

Reply:
I wonder why we did it that way. :confused:

Reply:
I think it was a bit of a wrong approach to testing the right thing. What the test really wanted to check is that no copying happened unnecessarily... so it probably should have checked for `vol.data is V._prev_data.data`, when `copy=False` and no clim-induced changes occur.

Reply:
We should try to avoid checking private properties of Visuals for validity. Checking that the array hasn't changed values is probably good enough for now.